r/src/filesystem.cpp

```diff
@@ -429,8 +429,8 @@ std::shared_ptr<fs::GcsFileSystem> fs___GcsFileSystem__Make(bool anonymous,
   }
 
   auto io_context = MainRThread::GetInstance().CancellableIOContext();
-  // TODO(ARROW-16884): update when this returns Result
-  return fs::GcsFileSystem::Make(gcs_opts, io_context);
+
+  return ValueOrStop(fs::GcsFileSystem::Make(gcs_opts, io_context));
```

Comment:
The R bindings supports both of old Arrow C++ and new Arrow C++:

```suggestion
#if ARROW_VERSION_MAJOR >= 19
  return ValueOrStop(fs::GcsFileSystem::Make(gcs_opts, io_context));
#else
  return fs::GcsFileSystem::Make(gcs_opts, io_context);
#endif
```

Reply:
Done! Ref b27686b4ccfb49171ca52596ab28734024696dcf (although I had to lower the major version to 18 to get workflow to finish).

Reply:
OK. Then let's wait for 18.0.0 release. We will bump development version to 19 in the 18.0.0 release process.
FYI: You can monitor https://lists.apache.org/thread/c3t7wn32pp506925fm1bg15lrrk4jrz6 thread for the 18.0.0 release.

Reply:
Will monitor! Also removed unnecessary diff (ref 1d6b53ae38bfce51d828a0a473601e538dd1a4b0).